pythia/tasks/captioning/coco/dataset.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import os
+
+import torch
+import tqdm
+
+from pythia.common.sample import Sample
+from pythia.tasks.base_dataset import BaseDataset
+from pythia.tasks.features_dataset import FeaturesDataset
+from pythia.tasks.image_database import ImageDatabase
+from pythia.utils.distributed_utils import is_main_process
+from pythia.utils.general import get_pythia_root
+
+
+class COCODataset(BaseDataset):
```

Comment:
It probably will be better to inherit VQA2Dataset class here and use most of the similar code from it.